botState.go

```diff
@@ -113,5 +116,14 @@ func (bs *BotState) ParsePlayerData(playerName, key, value string) error {
 }
 
 func (bs *BotState) AvailableMoves() []*Move {
+	var movesWithKo []*Move = bs.field.AvailableMoves()
+	var moves []*Move = make([]*Move, 0)
+	for _, move := range movesWithKo {
+		var currentHashForMove uint32 = bs.field.HashAt(move.X, move.Y)
+		_, has := bs.history[bs.field.zhash^currentHashForMove^zhashes[move.X][move.Y][1]]
+		if !has {
+			moves = append(moves, move)
+		}
```

Comment:
Abstract the ko detection into its own function. We're probably going to want to use it in our evaluation in the future instead of just for filtering illegal moves.